types/firestore.d.ts

```diff
@@ -1616,7 +1659,7 @@ declare namespace FirebaseFirestore {
      * @param snapshot The snapshot of the document to start after.
      * @return The created Query.
      */
-    startAt(snapshot: DocumentSnapshot<any>): Query<T>;
+    startAt(snapshot: DocumentSnapshot): Query<AppModelType, DbModelType>;
```

Comment:
startAt(), startAfter(), endAt(), and endBefore() should be parameterized rather than hardcoding the default type parameters for the given DocumentSnapshot.